src/Draco.Compiler/Internal/Semantics/AbstractSyntax/AstBuilder.cs

```diff
@@ -34,7 +34,15 @@ internal static class AstBuilder
     /// <param name="db">The <see cref="QueryDatabase"/> for the computation.</param>
     /// <param name="decl">The <see cref="ParseNode"/> to construct the <see cref="Ast"/> from.</param>
     /// <returns>The <see cref="Ast"/> form of <paramref name="decl"/>.</returns>
-    public static Ast.Decl ToAst(QueryDatabase db, ParseNode.Decl decl) => db.GetOrUpdate(
+    public static Ast.Decl ToAst(QueryDatabase db, ParseNode.Decl decl)
+    {
+        // Get all the doc commemts above the declarationh
+        var trivia = decl.Tokens.FirstOrDefault() is not null ?
+            decl.Tokens.FirstOrDefault()!.LeadingTrivia.Where(x => x.Type == TriviaType.DocumentationComment) :
+            null;
+        // Concatenate the text of all the doc comments
+        var documentation = trivia is not null ? string.Join(Environment.NewLine, trivia.Select(x => x.Text.TrimStart('/'))) : null;
```

Comment:
You moved logic out of the memoization context, I suggest not doing that.
Also, the details of this is a bit fuzzy for me, let's discuss how we actually want to extract documentation blocks. And definitely have it as a separate method, let's not litter it here.